src/App.tsx

```diff
@@ -1,57 +1,66 @@
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Dropdown } from './components/Dropdown';
+import { Person } from './types/Person';
+
+const getFilteredPeople = (query:string) => {
+  if (!query) {
+    return peopleFromServer;
+  }
+
+  const newPeople = peopleFromServer
+    .filter((person) => person.name.toLowerCase()
+      .includes(query.toLowerCase()));
+
+  return newPeople;
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selected, setSelected] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const people = getFilteredPeople(appliedQuery);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 500),
+    [appliedQuery],
+  );
+
+  const onQueryChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(event.target.value);
+      setSelected(null);
+      applyQuery(event.target.value);
+    },
+    [applyQuery],
+  );
+
+  const onSelected = useCallback((person: Person | null) => {
+    if (person) {
+      setSelected(person);
+      setQuery('');
+      applyQuery('');
+    }
+  }, [applyQuery]);
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
-
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
-
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
-        </div>
-      </div>
+      {selected ? (
+        <h1 className="title">{`${selected?.name} (${selected?.born} = ${selected?.died})`}</h1>
+      ) : (
+        <h1 className="title">Select a person</h1>
+      )}
```

Comment:
let's write one `h1` tag and only change children

Reply:
fixed